File: src/IsbInterop/Presentation/Proxies/ActionList.vb
Imports IsbInterop.Accessory.Proxies

Namespace Presentation.Proxies

  ''' <summary>
  ''' Обертка над IActionList.
  ''' </summary>
  Friend Class ActionList
    Inherits List(Of IAction)
    Implements IActionList

    ''' <summary>
    ''' Получает действие по имени.
    ''' </summary>
    ''' <param name="name">Имя действия.</param>
    ''' <returns>Действие из списка с именем name, если оно существует, иначе null.</returns>
    Public Function FindAction(name As String) As IAction Implements IActionList.FindAction
      Dim actionRcw = InvokeRcwInstanceMethod("FindAction", name)
      Return If(actionRcw IsNot Nothing, New Action(actionRcw, Scope), Nothing)
    End Function

    ''' <summary>
    ''' Конструктор.
    ''' </summary>
    ''' <param name="rcwIActionList">COM-объект IActionList.</param>
    ''' <param name="scope">Область видимости.</param>
    Public Sub New(rcwIActionList As Object, scope As IScope)
      MyBase.New(rcwIActionList, scope)
    End Sub
  End Class
End Namespace

File: src/IsbInterop/Presentation/Proxies/ActionList.cs
using IsbInterop.Accessory.Proxies;
using IsbInterop.Collections.Proxies;

namespace IsbInterop.Presentation.Proxies
{
  /// <summary>
  /// Обертка над IActionList.
  /// </summary>
  internal class ActionList : List<IAction>, IActionList
  {
    /// <summary>
    /// Получает действие по имени.
    /// </summary>
    /// <param name="name">Имя действия.</param>
    /// <returns>Действие из списка с именем name, если оно существует, иначе null.</returns>
    public IAction FindAction(string name)
    {
      var actionRcw = InvokeRcwInstanceMethod("FindAction", name);
      return actionRcw != null ? new Action(actionRcw, Scope) : null;
    }

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="rcwIActionList">COM-объект IActionList.</param>
    /// <param name="scope">Область видимости.</param>
    public ActionList(object rcwIActionList, IScope scope) : base(rcwIActionList, scope) { }
  }
}
